Nodejs/Product/Nodejs/Project/NodejsFileNode.cs

```diff
@@ -38,6 +39,16 @@ public NodejsFileNode(NodejsProjectNode root, ProjectElement e)
             }
         }
 
+        protected override void OnParentSet(HierarchyNode parent) {
+#if DEV14
+            if (Url.EndsWith(".d.ts", StringComparison.OrdinalIgnoreCase) && Url.IndexOf(@"\typings\", StringComparison.OrdinalIgnoreCase) >= 0) {
+                ProjectMgr.Site.GetUIThread().Invoke(() => {
+                    this.IncludeInProject(true);
+                });
```

Comment:
is it necessary to invoke the UI thread here? Let's double check before merging in.